server/internal/server/vector_stores.go

```diff
@@ -0,0 +1,321 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	fv1 "github.com/llm-operator/file-manager/api/v1"
+	"github.com/llm-operator/rbac-manager/pkg/auth"
+	v1 "github.com/llm-operator/vector-store-manager/api/v1"
+	"github.com/llm-operator/vector-store-manager/server/internal/store"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+const (
+	vectorStoreObject = "vector_store"
+)
+
+// CreateVectorStore creates a new vector store.
+func (s *S) CreateVectorStore(
+	ctx context.Context,
+	req *v1.CreateVectorStoreRequest,
+) (*v1.VectorStore, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Maybe follow OpenAI and validate the metadata. 

> Set of 16 key-value pairs that can be attached to an object. This can be useful for storing additional information about the object in a structured format. Keys can be a maximum of 64 characters long and values can be a maxium of 512 characters long.

https://platform.openai.com/docs/api-reference/vector-stores/modify

Btw, seems like there is a typo in the page (maxium -> maximum).